src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -920,6 +1240,29 @@ otError Leader::RemoveRloc(uint16_t aRloc16)
             break;
         }
 
+#if OPENTHREAD_ENABLE_SERVICE
+
+        case NetworkDataTlv::kTypeService:
+        {
+            if (!aOnlyRouter)
```

Comment:
I still could not figure out why "SRV stuff should stay in place" when `RemoveBorderRouter()`.  
If the leader decides to release the router id, it means the router might be not available in the mesh network any longer,   then what's the point of keeping the service information of this router in the network data? Would you please help to explain it more?

Reply:
Actually, you're right. I wasn't analyzing the MleRouter code well enough. Cleaning it back.